langchain4j-vertex-ai/src/main/java/dev/langchain4j/model/vertexai/VertexAiRankerScoringModel.java

```diff
@@ -0,0 +1,144 @@
+        this.location = ensureNotBlank(location, "location");
+        this.model = model;
+        this.titleMetadataKey = titleMetadataKey;
+    }
+
+    /**
+     * Scores all provided {@link TextSegment}s against a given query.
+     *
+     * @param segments The list of {@link TextSegment}s to score.
+     * @param query    The query against which to score the segments.
+     * @return the list of scores. The order of scores corresponds to the order of {@link TextSegment}s.
+     */
+    @Override
+    public Response<List<Double>> scoreAll(List<TextSegment> segments, String query) {
+        AtomicInteger counter = new AtomicInteger();
+
+        try (RankServiceClient rankServiceClient = RankServiceClient.create(
+            RankServiceSettings.newBuilder().build())) {
+
+            RankRequest.Builder rankingRequestBuilder = RankRequest.newBuilder();
+
+            if (model != null && !model.isEmpty()) {
+                rankingRequestBuilder.setModel(model);
+            }
+
+            rankingRequestBuilder
+                .setRankingConfig(RankingConfigName.newBuilder()
+                    .setProject(projectId)
+                    .setLocation(location)
+                    .setRankingConfig(
+                        String.format("projects/%s/locations/%s/rankingConfigs/default_ranking_config.", projectNum, location))
+                    .build().getRankingConfig())
+                .setQuery(query)
+                .setIgnoreRecordDetailsInResponse(true)
+                .addAllRecords(segments.stream()
+                    .map(segment -> {
+                        RankingRecord.Builder rankingBuilder = RankingRecord.newBuilder()
+                            .setContent(segment.text());
+                        // Ranker API takes into account titles in its score calculations
+                        String titleKeyToUse = titleMetadataKey != null ? titleMetadataKey : "title";
+                        if (segment.metadata().getString(titleKeyToUse) != null) {
+                            rankingBuilder.setTitle(segment.metadata().getString(titleKeyToUse));
+                        }
+                        // custom ID used to reorder the (sorted) results back into original segment order
+                        rankingBuilder.setId(String.valueOf(counter.getAndIncrement()));
+                        return rankingBuilder.build();
+                    })
+                    .collect(Collectors.toList()));
+
+            RankResponse rankResponse = rankServiceClient.rank(rankingRequestBuilder.build());
+
+            return Response.from(rankResponse.getRecordsList().stream()
+                .peek(rankingRecord -> {
+                    System.out.println(rankingRecord.toString());
+                })
+                // the API returns results sorted by relevance score, so reorder them back to original order
+                .sorted((rr1, rr2) -> (Double.valueOf(rr1.getId()) > Double.valueOf(rr2.getId())) ? 1 : -1)
```

Comment:
```suggestion
                .sorted(comparing(RankingRecord::getId))
```

Reply:
But using `comparing` that way, it's going to compare strings, as `getId()` returns strings.